### PR TITLE
Add 'enabled' field to pulse channels, use to enable/disable without forgetting settings

### LINKED ISSUE #2219: Allow pulses to be temporarily turned off
We should give the user a way to temporarily turn off a pulse without deleting it. Simplest way with no UI changes would be to relax the >= 1 pulse channel requirement for existing pulses.

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #2219

@agilliland @camsaul you guys should be sure to review my Clojure/migration changes.

### CODE REVIEW

File: resources/migrations/033_add_pulse_channel_enabled_field.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+databaseChangeLog:
+  - changeSet:
+      id: 33
```
Comment (agilliland) [inline resources/migrations/033_add_pulse_channel_enabled_field.yaml]: the PR i merged yesterday used id 33, so you'll need to bump this up to 34 and rename the file.

File: frontend/src/pulse/components/PulseEditChannels.jsx [hunk 2]
```diff
@@ -183,14 +193,14 @@ export default class PulseEditChannels extends Component {
     renderChannelSection(channelSpec) {
         let { pulse, user } = this.props;
         let channels = pulse.channels
-            .map((c, index) => c.channel_type === channelSpec.type ? this.renderChannel(c, index, channelSpec) : null)
-            .filter(e => !!e);
+            .map((c, i) => [c, i]).filter(([c, i]) => c.enabled && c.channel_type === channelSpec.type)
```
Comment (agilliland) [inline frontend/src/pulse/components/PulseEditChannels.jsx]: I'm not sure if this makes sense.  if the channels on a pulse are disabled then there is no way to access the pulse and edit it any more.

Comment (tlrobinson) [inline frontend/src/pulse/components/PulseEditChannels.jsx]: `toggleChannel` re-enables existing channels of the matching type.

**Comment by agilliland** [comment]
does it delete the pulse if you toggle off all the channels?  that's the case i'm unclear about because i think it's filtering out all pulses with no active channels from the pulse listing, which means that if you have a pulse with only 1 channel in it and you disable that channel, then you'll lose any way to get back to that pulse.

**Comment by tlrobinson** [comment]
It's never filtering pulses, only pulse channels. I figured we shouldn't say "Emailed to ..." or "Slack'd ..." in the pulse listing if that channel is disabled.

File: resources/migrations/034_add_pulse_channel_enabled_field.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+databaseChangeLog:
+  - changeSet:
+      id: 33
```
Comment (agilliland) [inline resources/migrations/034_add_pulse_channel_enabled_field.yaml]: lets change this id to match the file as well

File: frontend/src/pulse/components/PulseEditChannels.jsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import React, { Component, PropTypes } from "react";
 
 import _ from "underscore";
+import i from "icepick";
```
Comment (camsaul) [inline frontend/src/pulse/components/PulseEditChannels.jsx]: question.

If I have a cool loop with `i` as my index variable I will inadvertently end up shadowing the icepick `i` variable. Then if I try to do something like `i.assocIn` I'll get a `assocIn is not a function`, breaking all of the Metabase.

I would posit that reserving a common variable name like `i` is bound to lead to just such chaos. Why would the library writers do such a thing to us? Is there any way we can import it as an alias?

File: src/metabase/models/pulse_channel.clj [hunk 5]
```diff
@@ -158,18 +158,19 @@
...
-                        (= :schedule_day weekday))
-                   (and (= :schedule_type (name :monthly))
-                        (= :schedule_hour hour)
-                        (= :schedule_frame schedule-frame)
-                        (or (= :schedule_day weekday)
-                            ;; this is here specifically to allow for cases where day doesn't have to match
-                            (= :schedule_day monthly-schedule-day-or-nil))))))))
+      (k/where (and (= :enabled true)
+                    (or (= :schedule_type (name :hourly))
+                        (and (= :schedule_type (name :daily))
```
Comment (camsaul) [inline src/metabase/models/pulse_channel.clj]: I just noticed this, don't know how long it's been this way. But `(name <keyword>)` is just the string name of that keyword. e.g.

```clojure
(name :daily) -> "daily"
```

We should just write the string and save some CPU cycles.

```clojure
(= :schedule_type "monthly")
```

File: frontend/src/pulse/components/PulseEditChannels.jsx (hunk 4 above)
Comment (tlrobinson) [inline frontend/src/pulse/components/PulseEditChannels.jsx]: Yes, you can give it whatever alias you like, or do `import { assocIn } from "icepick";`

But with `map` and `for (... of ...)` loops it's become a lot less common to write `for (i = 0;...)` style loops in JS anyway.

**Pull request merged**